router/qlog/provider/local_test.go

```diff
@@ -0,0 +1,52 @@
+package qlog
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDumpQueryFileExists(t *testing.T) {
+	assert := assert.New(t)
+
+	tmpFile, err := os.CreateTemp("", "dump-test")
+	assert.NoError(err, "")
+	defer func() {
+		err = os.Remove(tmpFile.Name())
+		assert.NoError(err, "")
+	}()
+
+	testStr := "ogima testing localQLog"
+	localQLog := NewLocalQlog()
+	err = localQLog.DumpQuery(context.TODO(), tmpFile.Name(), testStr)
+
+	assert.NoError(err, "")
+
+	data, err := os.ReadFile(tmpFile.Name())
+	assert.NoError(err)
+
+	assert.Equal(fmt.Sprintf("%s\n", testStr), string(data))
+}
+
+func TestDumpQueryFileNotExist(t *testing.T) {
```

Comment:
Can you also add tests with error on openning log file(non-writable file, for example)?